plugin/src/withSpotlight.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  ConfigPlugin,
+  createRunOncePlugin,
+  WarningAggregator,
+  withAppDelegate,
+} from "@expo/config-plugins";
+
+const pkg = require("react-native-spotlight-search/package.json");
+
+const SPOTLIGHT_IMPORT = `#import "RCTSpotlightSearch.h"
+`;
+const SPOTLIGHT_ACTIVITY = `- (BOOL)application:(UIApplication *)application continueUserActivity:(nonnull NSUserActivity *)userActivity restorationHandler:(nonnull void (^)(NSArray<id<UIUserActivityRestoring>> * _Nullable))restorationHandler {`;
+
+const SPOTLIGHT_ACTIVITY_ADD = `- (BOOL)application:(UIApplication *)application continueUserActivity:(nonnull NSUserActivity *)userActivity restorationHandler:(nonnull void (^)(NSArray<id<UIUserActivityRestoring>> * _Nullable))restorationHandler {
+  [RCTSpotlightSearch handleContinueUserActivity:userActivity];
+  return YES;`;
+
+const modifyAppDelegate = (appDelegate: string) => {
+  if (!appDelegate.includes(SPOTLIGHT_IMPORT)) {
+    appDelegate = SPOTLIGHT_IMPORT + appDelegate;
+  }
+  const updatedAppDelegate = appDelegate.replace(
+    SPOTLIGHT_ACTIVITY,
+    SPOTLIGHT_ACTIVITY_ADD
+  );
+  return updatedAppDelegate;
+};
+
+const withSpotlightAppDelegate = (config: any) => {
+  return withAppDelegate(config, (config) => {
+    if (config.modResults.language === "objc") {
```

Comment:
This is going to break in React Native 68 since they use `AppDelegate.mm`, or `objcpp`. Please check out the Expo Modules Core AppDelegate Subscriber pattern to implement a safe version of this, could even drop the config plugin altogether. Here is an [example](https://github.com/react-native-google-signin/google-signin/pull/1033), [here are the docs](https://docs.expo.dev/modules/appdelegate-subscribers/).